test/1_stdlib/ErrorHandling.swift

```diff
@@ -340,4 +340,42 @@ ErrorHandlingTests.test("ErrorHandling/Collection map") {
   }
 }
 
+/* when b == 2 we are losing elements. insertion sort throws error
+in between while sorting so loop is not able to complete and swap the elements and array mutates
+to [2,3,3]
+ErrorHandlingTests.test("ErrorHandling/sort") {
+  var collection = [3, 2, 1]
+  do {
+    try collection.sort { (a, b) throws -> Bool in
+      print("collection before error \(collection)")
+      print("a \(a) \(b)")
+
+      if b == 2 {
```

Comment:
on Thu Apr 21 2016, Dmitri Gribenko <notifications-AT-github.com> wrote:

>
> I find losing elements surprising though. Wouldn't we want to ensure that we
> don't lose elements as a matter of QoI for this API specifically?

No, trying to do that is a losing battle and there's nothing useful the
user can do with the result anyway.

-- 
Dave